src/common/Requirements.ts

```diff
@@ -123,82 +119,87 @@ export abstract class BaseSetup implements RequirementList {
     }
 
     public async executeSetup(): Promise<SetupTestResult> {
-        const allPromises: Array<Promise<any>> = [];
-        this.requirements.forEach((requirement) =>
-            allPromises.push(WrappedPromise(requirement))
+        const testResult: SetupTestResult = {
+            hasMetAllRequirements: true,
+            tests: []
+        };
+
+        let totalDuration = 0;
+        const setupTasks = new Listr(
+            [
+                {
+                    task: (rootCtx, rootTask): Listr => {
+                        const subTasks = new Listr([], {
+                            concurrent: true,
+                            rendererOptions: { collapse: false }
+                        });
+                        for (const requirement of this.requirements) {
+                            subTasks.add({
+                                options: { persistentOutput: true },
+                                task: (subCtx, subTask): Promise<void> =>
+                                    WrappedPromise(requirement).then(
+                                        (result) => {
+                                            testResult.tests.push(result);
+                                            if (!result.hasPassed) {
+                                                testResult.hasMetAllRequirements = false;
+                                            }
+
+                                            subTask.title = this.getFormattedTitle(
```

Comment:
we'll start with `requirement.title` as the task title, but will update the title (with colored passed/failed messages) after a task is completed.